qa/sc_csw_in_fee_payment.py

```diff
@@ -0,0 +1,317 @@
+                    ", actual: " + format(vsc_ccout_["vFieldElementCertificateFieldConfig"]))
+        assert_true("wCeasedVk" in vsc_ccout_, "CSW verification key should be present")
+
+        # Checks that Sidechain creation transaction is in SC block
+        mc_block_hash = mc_sc_creation_tx["blockhash"]
+        sc_block_id = sc_node.block_best()["result"]["block"]["id"]
+        check_mcreference_presence(mc_block_hash, sc_block_id, sc_node)
+
+        # ******************** EPOCH 0 START ********************
+        epoch_mc_blocks_left = self.sc_withdrawal_epoch_length - 1
+
+        # create 1 FTs in the same MC block to SC
+        sc_address_1 = sc_node.wallet_createPrivateKey25519()["result"]["proposition"]["publicKey"]
+        ft_amount_1 = 100
+        mc_return_address_1 = mc_node.getnewaddress()
+
+        forward_transfer_to_sidechain(self.sidechain_id, mc_node,
+                                      sc_address_1, ft_amount_1, mc_return_address_1)
+
+        # Sleep for 1 second to let MC synchronize wallet
+        time.sleep(1)
+
+        epoch_mc_blocks_left -= 1
+
+        # Generate 1 SC block to include FTs.
+        generate_next_blocks(sc_node, "first node", 1)[0]
+
+        # Spend the FT, send it to ourselves.
+        amount = 20
+        sc_address_2 = sc_node.wallet_createPrivateKey25519()["result"]["proposition"]["publicKey"]
+        sendCoinsToAddress(sc_node, sc_address_2, amount, self.FEE)
+
+        # Generate 1 SC block to include Tx.
+        generate_next_blocks(sc_node, "first node", 1)[0]
+
+        # Create another transaciton to ourselves.
+        sc_address_3 = sc_node.wallet_createPrivateKey25519()["result"]["proposition"]["publicKey"]
+        amount = 30
+        sendCoinsToAddress(sc_node, sc_address_3, amount, self.FEE)
+
+        # Generate 1 SC block to include Tx.
+        generate_next_blocks(sc_node, "first node", 1)[0]
+
+        # Generate 8 more MC blocks to finish the first withdrawal epoch, then generate 1 more SC block to sync with MC.
+        we0_end_mcblock_hash = mc_node.generate(epoch_mc_blocks_left)[-1]
+        print("End mc block hash in withdrawal epoch 0 = " + we0_end_mcblock_hash)
+        we0_end_mcblock_json = mc_node.getblock(we0_end_mcblock_hash)
+        we0_end_epoch_cum_sc_tx_comm_tree_root = we0_end_mcblock_json["scCumTreeHash"]
+        print("End cum sc tx cum comm tree root hash in withdrawal epoch 0 = " + we0_end_epoch_cum_sc_tx_comm_tree_root)
+        sc_block_id = generate_next_block(sc_node, "first node")
+        check_mcreferencedata_presence(we0_end_mcblock_hash, sc_block_id, sc_node)
+
+        # Verify that we have FeePayment Box in the last block and in the wallet
+        block = http_block_findById(sc_node, sc_block_id)
+        assert_false(block["block"]["header"]["feePaymentsHash"] == "0000000000000000000000000000000000000000000000000000000000000000")
+
+        fee_payments = http_block_getFeePayments(sc_node, sc_block_id)
```

Comment:
more safe to extract the boxid from fee_payments and use it below to search inside all_boxes instead of using the value

Reply:
Right, thanks